__tests__/ForgotPassword.js

```diff
@@ -0,0 +1,320 @@
+import '@testing-library/jest-dom/extend-expect';
+
+import { fireEvent, wait, waitForDomChange } from '@testing-library/react';
+import { forceRemoveAll } from 'lib/flashMessage';
+import RoutePath from 'lib/routePath';
+import nock from 'nock';
+import { StatusCodes } from 'shared/constants';
+import { AppRoutes } from 'shared/constants/routes';
+import {
+  authTokenResponse,
+  AWSInfoResponse,
+  generateRandomString,
+  getMockCall,
+  postMockCall,
+  USER_EMAIL,
+  userResponse,
+} from 'testUtils/mockHttpCalls';
+import { renderRouteWithStore } from 'testUtils/renderUtils';
+
+describe('PasswordReset', () => {
+  beforeAll(() => {
+    nock.disableNetConnect();
+  });
+
+  afterAll(() => {
+    nock.enableNetConnect();
+  });
+
+  afterEach(async () => {
+    if (!nock.isDone()) {
+      // eslint-disable-next-line no-console
+      console.error('Nock has pending mocks:', nock.pendingMocks());
+    }
+
+    await wait(() => expect(nock.isDone()).toBeTruthy());
```

Comment:
Sorry, I'm being late again.

One thing I am seeing now in the tests I am refactoring is that this pattern (console.error and then await fro the expect) will trigger a lot of false positives, it will log a lot of `console.error`s that shouldn't be logged, nor consiered as errors. 

If we expect here to all nocks to be done, a lot of async actions will appear in the _Nock has pendings mock: _ message, and then you might think (I did) that you are intercepting too many calls, but what happens is that these calls are not yet matched, which is something expected actually. 

The `console.error` makes sense when debugging, if you want to find out which calls you are intercepting but happa is not using. But this two snippets together doesn't make sense and can lead to confusion IMO.

Reply:
Yeah, you're right. Actual checking for `isDone()` for printing pending mocks should be in the `wait` callback. Guess it's too late now 💣

Reply:
It's ok like this, it works, it's just that we might get too many logs and that can be pretty confusing, especially when some tests are failing and you are debugging. Just something to be aware of.